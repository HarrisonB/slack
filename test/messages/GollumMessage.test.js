const { GollumMessage } = require('../../lib/messages/gollum');

const gollum = require('../fixtures/webhooks/gollum.json');

describe('GollumMessage rendering', () => {
  test('works', async () => {
    const gollumMessage = new GollumMessage({
      gollum,
    });
    const rendered = gollumMessage.getRenderedMessage();
    expect(rendered).toMatchSnapshot();
  });
});
