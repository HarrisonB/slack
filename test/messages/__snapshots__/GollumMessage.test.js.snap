// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GollumMessage rendering works 1`] = `
Object {
  "attachments": Array [
    Object {
      "author_icon": "https://avatars2.githubusercontent.com/u/9831992?v=4",
      "author_link": "https://github.com/rachmari",
      "author_name": "rachmari",
      "color": "#24292f",
      "fallback": "[Codertocat/Hello-World] 2 wiki pages changed",
      "footer": "<https://github.com/Codertocat/Hello-World|Codertocat/Hello-World>",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "mrkdwn_in": Array [
        "text",
      ],
      "text": "*2 wiki pages changed*
edited <https://github.com/Codertocat/Hello-World/wiki/Home|Home>
created <https://github.com/Codertocat/Hello-World/wiki/Another-Page|Another Page>",
    },
  ],
}
`;
